examples_test.go

```diff
@@ -10,7 +10,7 @@ import (
 // flow support is globally available, but enables logging in to hosted GitHub instances as well.
 func Example() {
 	flow := &Flow{
-		Hostname:     "github.com",
+		Server:       ServerGitHub("github.com"),
```

Comment:
Could Server be an interface rather than a struct? That way the caller can pass in any object that responds to the set of `*URL()` methods that we need. Additionally, if the Server is nil, we can fall back to `Hostname` and thus ensure backwards compatibility with existing calls.

Reply:
I would have no problem with doing this - but I'm curious why? IMO being an interface makes implementation more tedious for end-users, and doesn't really present any major benefits here that I can see.

As for backwards compatibility, that could also be solved by using a pointer to `Server`.